webhooks/cdap_webhook.go

```diff
@@ -0,0 +1,127 @@
+package webhooks
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+
+	v1alpha1 "cdap.io/cdap-operator/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const (
+	labelInstanceKey           = "cdap.instance"
+	// cdapMasterNamespaceKey is the label added by CDAP on pods launched by CDAP.
+	cdapMasterNamespaceKey     = "cdap.k8s.namespace"
+	// customResourceNamespaceKey is the label added by controller-reconciler on resources
+	// managed by the CDAP operator.
+	customResourceNamespaceKey = "custom-resource-namespace"
```

Comment:
Is this the namespace of the CDAPMaster for a resource managed by the operator?

Reply:
`custom-resource-namespace` is added to pods managed by the operator. It points to the namespace of the CDAP CR. Here is the responsible code: https://github.com/cdapio/cdap-operator/blob/develop/vendor_old/sigs.k8s.io/controller-reconciler/pkg/genericreconciler/handler.go#L62-L70

I've verified these labels are present in pods launched directly by the operator (not through twill).